### DISCUSSION TITLE
Additional parameters/properties , how to configure

### DISCUSSION THREAD

**Discussion by veerur4u** [jomjol/AI-on-the-edge-device#2617] [General] [answered]
How to pass/configure additional properties/parameters pertaining to the Meter. For example , we would like send MeterId, Location, Building Name etc. How to configure these and receive same on the MQTT endpoint. 

Thanks

**Comment by caco3** [comment]
You can set the MQTT main topic and also the meter name.
See https://jomjol.github.io/AI-on-the-edge-device-docs/Parameters/#parameter-maintopic and https://jomjol.github.io/AI-on-the-edge-device-docs/ROI-Configuration/
Beside of those fields there is no functionality for this.

**Reply by veerur4u** [reply]
@caco3 . Thanks for response, if I understood correctly ROI is more for configuring the Digital and Analog values that are captured from the Meter. The values/properties (MeterId and BuildingName ), I'm referring too are more of static ones for each meter and they never change, however we need these values to be passed across each message so that the MQTT listener will identify which Meter values are being received . 
For example, if there are 10 meters , all of the 10 meters will publish message to same main topic (e568b626dd3c4b8caf94d0b91da5d7f3.s1.eu.hivemq.cloud:8883/watermeter/main/json) [**{"value": "521.17108", "raw": "521.17108", "pre": "521.17108", "error": "no error", "timestamp": "2023-01-13T16:00:42+0100" }**]
We would like to pass MeterId and BuildingName along with this.  Please suggest

**Reply by caco3** [reply]
You could set the Meter ID as the `maintopic`, then it is the first part of your MQTT topi, eg. `watercounter/main/json`.

You also could set the maintopic to `<MeterID>__<Building>` if you want.

Also, the number name in the ROI settings (usually `main`) can be set to whatever you want. This also is directly represented in teh MQTT topic.

**Reply by veerur4u** [reply]
The problem with setting MeterId in main topic, makes it dynamic and also we end up with lots of topics.  We are planning to use one main  Topic (for all Meters) and that will have a listener which then process the data. Below diagram depicts the flow. The question is can we pass some additional parameter/properties (Which are not displayed/available on the Meter to read, but can be manually configured ) 

<img width="594" alt="image" src="https://github.com/jomjol/AI-on-the-edge-device/assets/26367561/8347cceb-f44b-4206-a8fb-7dc58805c230">

**Reply by caco3** [reply]
The maintopic is just a string. it can also contain slashs to divide it further, eg. `test/aa/bb/cc/wasserzaehler`:
![grafik](https://github.com/jomjol/AI-on-the-edge-device/assets/1783586/a2f615e4-5c20-424a-93ad-0256ded9039c)

**Reply by caco3** [reply, marked as answer]
I added a note to the docs, see https://jomjol.github.io/AI-on-the-edge-device-docs/Parameters/#parameter-maintopic
